### PR TITLE
added burst plotting script

### PR DESCRIPTION
**Pull request by otirmizi**

### CODE REVIEW

**Comment by otirmizi** [comment]
The new script reads info from the SLC and outputs a .csv file containing burst IDs and other metadata. The script also outputs an interactive html file of the plotted bursts as well as a shapefile of the GeoDataframe

File: src/s1reader/utils/plot_bursts.py
Comment (hfattahi) [inline src/s1reader/utils/plot_bursts.py:89]: If epsg is 4326, the variation is small and results in identical x_min x_max and identical y_min y_max.

**Review by hfattahi — COMMENTED** [review]
Thanks for the PR. It plots a very nice map. Would be nice if we extend to multiple zip files. Or if user gives the path to the directory with tons of zip files and date of interest and the script goes and finds all the zip files related to that date and then makes the plot. It is ok with me if we merge the current PR and then add the above functionalities later.

**Review by hfattahi — APPROVED** [review]
LGTM. Congrats on your first PR. Nice work.

**Comment by hfattahi** [comment]
Sorry for commenting again even though I approved. It seems like kml comes for free with what you already have. All you need to do is:

```
import fiona
fiona.supported_drivers['KML'] = 'rw'
gdf2.to_file('burst_map.kml', driver='KML')
```
That would make the kml which you can open in google earth. Here is a screenshot:
Screen Shot 2022-05-24 at 12 54 15 PM

Do you mind adding those few lines.

File: src/s1reader/utils/plot_bursts.py
Comment (LiangJYu) [inline src/s1reader/utils/plot_bursts.py:12]: pandas, geopandas, folium, and fiona are currently not part module requirements. I'm not sure how to proceed. AFAICT there's no need to require them for the production environment. @hfattahi what do you think?

Wrap `try-except`?

Do something like [ISCE3](https://github.com/isce-framework/isce3/tree/develop/.ci/images/centos), where we have different requirement files for documentation sake?

File: src/s1reader/utils/plot_bursts.py
Comment (LiangJYu) [inline src/s1reader/utils/plot_bursts.py:43]: Please add docstring. [example](https://github.com/opera-adt/s1-reader/blob/59895266c296ee26ee7e171534c1ffe989052304/src/s1reader/s1_burst_slc.py#L16-L42)

File: src/s1reader/utils/plot_bursts.py
Comment (LiangJYu) [inline src/s1reader/utils/plot_bursts.py:88]:
```suggestion
            if epsg == 4326:
```
Parser already provides `epsg` as `int`.
Also formatting nitpick: `==` to ` == `.

File: src/s1reader/utils/plot_bursts.py
Comment (LiangJYu) [inline src/s1reader/utils/plot_bursts.py:113]:
```suggestion
    gdf.rename(columns={'border': 'geometry', inplace=True}).set_geometry('geometry')
```
No need for new object `gdf2` since `gdf` is never used again.

File: src/s1reader/utils/plot_bursts.py
Comment (LiangJYu) [inline src/s1reader/utils/plot_bursts.py:116]:
```suggestion
    gdf2.to_file(f'{output_filename}.shp')
```
Something like this to make all the outputs per run distinct?

Maybe change `output_filename` to `output_file_prefix` and hard code file extension accordingly in the f-string in write/output?

**Review by LiangJYu — COMMENTED** [review]
Nice work. A lot useful functionality packed in here. Few comments.

File: src/s1reader/utils/plot_bursts.py
Comment (hfattahi) [reply src/s1reader/utils/plot_bursts.py:12]: @LiangJYu For this PR, try-except is fine with me. Yes this is just extra and many users including isce3 modules and compass don't need it. So the requirement should be kept light and we should document that for plotting purpose those extra packages should be installed or have another requirement file. Perhaps you could do those clarifications after this PR is merged.

File: src/s1reader/utils/plot_bursts.py
Comment (hfattahi) [reply src/s1reader/utils/plot_bursts.py:116]: Good idea and then you could use the same for kml, html outputs

**Review by LiangJYu — APPROVED** [review]
LGTM

File: src/s1reader/utils/plot_bursts.py
Comment (LiangJYu) [inline src/s1reader/utils/plot_bursts.py:12]:
```suggestion
import argparse
import glob
from importlib import import_module
import sys

named_libs = [('fiona', 'fiona'), ('folium', 'folium'), ('geopandas', 'gpd'),
            ('pandas', 'pd')]
for (name, short) in named_libs:
    try:
        lib = import_module(name)
    except:
        print(sys.exc_info())
    else:
        globals()[short] = lib

import matplotlib.pyplot as plt
from osgeo import gdal, osr
from shapely.geometry import Polygon
from shapely import wkt

from s1reader.s1_orbit import get_orbit_file_from_list
from s1reader.s1_reader import load_bursts
```
[pep8 import order](https://peps.python.org/pep-0008/#imports)

[error handling multiple imports](https://stackoverflow.com/a/44829497)

File: src/s1reader/utils/plot_bursts.py
Comment (LiangJYu) [inline src/s1reader/utils/plot_bursts.py:139]: An error occurs here if KML file path provided already exists. Is there an overwrite option? That or add in the line above:
```python
kml_path = f'{output_filename}.kml'
if os.path.isfile(kml_path):
    os.remove(kml_path)
```

**Review by LiangJYu — COMMENTED** [review]
Apologies for premature approval. Few more changes.

File: src/s1reader/utils/plot_bursts.py
Comment (LiangJYu) [reply src/s1reader/utils/plot_bursts.py]: Disregard this comment. I was completely wrong.

File: src/s1reader/utils/plot_bursts.py
Comment (LiangJYu) [reply src/s1reader/utils/plot_bursts.py]: Please add lines in between import groups for readability. From pep8 import section linked above:

>
> Imports should be grouped in the following order:
> 
>   1. Standard library imports.
>   2. Related third party imports.
>   3. Local application/library specific imports.
> 
> **You should put a blank line between each group of imports.**
>

File: src/s1reader/utils/plot_bursts.py
Comment (LiangJYu) [inline src/s1reader/utils/plot_bursts.py:4]:
```suggestion
import argparse
import glob
from importlib import import_module
import os
```
Nitpick: alphabetical order for readability

**Review by LiangJYu — COMMENTED** [review]
Almost there. A couple more nitpicks to fix and then this PR is good to go.

**Comment by LiangJYu** [comment]
@otirmizi You also need to check the [Codacy report](https://app.codacy.com/gh/opera-adt/s1-reader/pullRequest?prid=9480224) at the bottom of the PR page if it flags anything. 
image

**Comment by LiangJYu** [comment]
> @otirmizi You also need to check the [Codacy report](https://app.codacy.com/gh/opera-adt/s1-reader/pullRequest?prid=9480224) at the bottom of the PR page if it flags anything.

Just because something is flagged doesn't mean it needs or can be fix/addressed. LMK if you have any questions.

File: src/s1reader/utils/plot_bursts.py
Comment (LiangJYu) [inline src/s1reader/utils/plot_bursts.py:8]:
```suggestion
named_libs = [('fiona', 'fiona'), ('geopandas', 'gpd'), ('pandas', 'pd')]
```
folium is unused

File: src/s1reader/utils/plot_bursts.py
Comment (LiangJYu) [inline src/s1reader/utils/plot_bursts.py:16]:
```suggestion

```
Trailing whitespace - 2nd issue in Codacy.

Based on your editor, trailing whitespaces can be automatically removed on save.
[VS Code](https://stackoverflow.com/questions/30884131/remove-trailing-spaces-automatically-or-with-a-shortcut)
[vim](https://stackoverflow.com/questions/356126/how-can-you-automatically-remove-trailing-whitespace-in-vim)

File: src/s1reader/utils/plot_bursts.py
Comment (LiangJYu) [inline src/s1reader/utils/plot_bursts.py:127]:
```suggestion
              y_max = y_spacing * round(max(tgt_y) / y_spacing)
```
Another trailing whitespace.

**Review by LiangJYu — COMMENTED** [review]
A few more nits

**Review by LiangJYu — APPROVED** [review]
Thanks for bearing with me. LGTM

**Comment by LiangJYu** [comment]
Hit "Squash and merge".
image

Maybe clean up nit-pick related bullets, then "Confirm squash and merge" and aoilà.
image

**Comment by otirmizi** [comment]
@LiangJYu I don't have write access so that option is not showing up for me :/

image

**Comment by LiangJYu** [comment]
> @LiangJYu I don't have write access so that option is not showing up for me :/
> 
> image

I'll merge this PR. I didn't know you lack write access. 🤷

**Pull request merged**
